plugin-server/src/cdp/cdp-api.ts

```diff
@@ -388,4 +393,76 @@ export class CdpApi {
                 error: 'Method not allowed',
             })
         }
+
+    private postMailjetWebhook =
+        () =>
+        async (req: express.Request & { rawBody?: Buffer }, res: express.Response): Promise<any> => {
+            const signature = req.headers['x-mailjet-signature'] as string
+            const timestamp = req.headers['x-mailjet-timestamp'] as string
+
+            if (!signature || !timestamp || !req.rawBody) {
+                return res.status(403).send('Missing required headers or body')
+            }
+
+            const payload = `${timestamp}.${req.rawBody.toString()}`
+            const hmac = crypto.createHmac('sha256', this.MAILJET_SECRET).update(payload).digest()
+
+            try {
+                const signatureBuffer = Buffer.from(signature, 'hex')
+                if (
+                    hmac.length !== signatureBuffer.length ||
+                    !crypto.timingSafeEqual(new Uint8Array(hmac), new Uint8Array(signatureBuffer))
+                ) {
+                    return res.status(403).send('Invalid signature')
+                }
+
+                // Track Mailjet webhook metrics
+                const event = req.body as MailjetEvent
+                const category = eventTypeToCategory[
+                    event.event as keyof typeof eventTypeToCategory
+                ] as AppMetric['category']
+
+                if (event) {
+                    const metric: AppMetric = {
+                        teamId: -1, //TODO: Get teamId from Mailjet additional config,
+                        pluginConfigId: -2, // -2 is hardcoded for mailjet
+                        category: category,
+                        successes: 1,
+                        // TODO: Losing this information in the metric for now
+                        // properties: {
+                        //     event_type: event.event,
+                        //     email: event.email,
+                        //     campaign_id: event.mj_campaign_id,
+                        //     custom_campaign: event.customcampaign,
+                        //     message_id: event.message_id,
+                        //     error: event.error,
+                        //     error_type: event.error_related_to,
+                        //     smtp_reply: event.smtp_reply,
+                        //     blocked: event.blocked,
+                        //     timestamp: event.time
+                        // }
+                    }
+
+                    await this.hub.appMetrics.queueMetric(metric)
```

Comment:
Lets not queue it - lets just log something and add a prometheus metric for now. We need way more other components to know where this should go properly so lets wait for those rather than add a bunch of code that we might never need